### DISCUSSION TITLE
Fitting the model using a subset of the data?

### DISCUSSION THREAD

**Discussion by codeanticode** [kingaa/pomp#148] [Q&A] [answered]
I was thinking of using a subset of the temporal data to fit my model, and sounds like the argument ```times``` in the [pomp function](https://kingaa.github.io/pomp/manual/pomp.html) would allow me to do that.

For instance, I have a data frame with 100 rows and a "time" column, the first time value is 1 and the last one 100. So I tried:

```
data %>% pomp(t0=1, times=1:50, time="time", ...
```

to use the first 50 time points, but gives me the following error: ```the length of ‘times’ does not match that of the data.```

But if I use ```times=1:100``` I  get the same error, although now times has the same length as the data. I guess I'm misinterpreting how to use the times argument. What would be the correct way to do this? Thank you!

**Comment by kingaa** [comment, marked as answer]
No, this isn't a valid use of `times`.  Better to do something like
```
data |>
  filter(time <= 50) |>
  pomp(t0=1,times="time",...)
```
Note that I am using the new, base-R pipe operator `|>`.  So excited about it!

**Reply by codeanticode** [reply]
Very cool, thanks Aaron! I see that I was confused about the times argument, specially because I was using ```time="time"``` for what seems the same purpose, and that one does still work? Or is it deprecated?

Now, would the pomp object still contain the entire data range or only the filtered values? I'm asking this because what I wanted to do is to fit the model on a subrange of the data and then run simulations on the entire range and plot the simulated data alongside the entire data, how can I do that?

In fact, I used the times argument in simulate, and it appeared to work as as initially thought (as the range of time values I want to ran the simulation for). Does the following code make sense, if not what would be the correct way to do it?

```
data |>
  filter(time <= 50) |>
  pomp(t0=1,times="time",...) -> mdl
...
mdl  %>%
  simulate(params=theta_opt, nsim=9, format="data.frame", include.data=TRUE, times=1:100) -> sim_data

sim_data %>% ggplot(aes(x=time, y=cases, ... )
```

**Reply by kingaa** [reply]
`time` was never a formal argument of `pomp`.  If used alone, it would match `times` using **R**'s partial matching feature.  If used as you did, i.e., in conjunction with `times`, it becomes an element of the [userdata](https://kingaa.github.io/pomp/manual/userdata.html).  Probably not what you intended.

Since `pomp` only gets the data downstream of the filtering operation, it only knows about the times through time 50.  If you want to simulate a longer (or different) set of times using the parameters you've estimated, you might do something like the following.
```
data |>
  filter(time <= 50) |>
  mif2(t0=1,times="time",...) -> mf

mf |>
  simulate(times=1:100,nsim=100,format="d") -> simdat
```
Two things to note:
1. It's not necessary to call `pomp` directly, since it's called internally by, in this case, `mif2` and, in general, by every elementary or estimation algorithm.
2. `simulate`, along with just about every other elementary or estimation algorithm, allows one to adjust the original pomp object on the fly.  In this case, this is demonstrated by the change of the `times`.
